### PR TITLE
[FIX] Ship ucrt, openmp and intel dlls for apps on windows

### PR DESCRIPTION
**Pull request by marehr**
#1778 ensures that the DLLs are included in the package

### CODE REVIEW

File: util/cmake/SeqAnBuildSystem.cmake [hunk 1]
```diff
@@ -426,6 +426,79 @@ macro (seqan_setup_install_vars APP_NAME)
...
+      "$ENV{ProgramFiles}/IntelSWTools/compilers_and_libraries_${v}/windows/redist"
+      "$ENV{${programfilesx86}}/IntelSWTools/compilers_and_libraries_${v}/windows/redist"
+    )
+  mark_as_advanced(INTEL${v}_REDIST_DIR)
+  set(INTEL${v}_FILES_DIR "${INTEL${v}_REDIST_DIR}/${CMAKE_INTEL_ARCH}/compiler")
+
+  if(NOT CMAKE_INSTALL_DEBUG_LIBRARIES_ONLY)
+    set(__install__libs
+      "${INTEL${v}_FILES_DIR}/libmmd.dll"
+      "${INTEL${v}_FILES_DIR}/libiomp5md.dll"
```
Comment (h-2) [inline util/cmake/SeqAnBuildSystem.cmake]: Can you make installation of this dll depend on `CMAKE_INSTALL_OPENMP_LIBRARIES` ?

File: util/cmake/SeqAnBuildSystem.cmake [hunk 2]
```diff
@@ -426,6 +426,79 @@ macro (seqan_setup_install_vars APP_NAME)
...
+
+macro (seqan_install_required_system_libraries)
+  # include intel dll's
+  if(COMPILER_WINTEL)
+    INTEL_FILES_FOR_VERSION(2016)
+  endif()
+
+  # The following include automates the MS Redistributable installer.
+  set (CMAKE_INSTALL_UCRT_LIBRARIES TRUE)
+  set (CMAKE_INSTALL_OPENMP_LIBRARIES TRUE)
```
Comment (h-2) [inline util/cmake/SeqAnBuildSystem.cmake]: Can you make this value be == `OPENMP_FOUND` and move it seven lines up?

**Comment by h-2** [comment]
AH, thats a very complete solution, you could probably create a PR to Kitware/cmake directly :)
I have proposed some minor changes so that we only install OpenMP libs when they are actually needed (OpenMP detection always happens before `seqan_configure_cpack_app` is called!).

**Comment by marehr** [comment]
For an official PR, I would need to investigate how the older versions of intel redistributed their dlls. If intel supports arm and where the path of the official [redistributable's](https://software.intel.com/en-us/articles/intelr-composer-redistributable-libraries-by-version) are.

**Comment by h-2** [comment]
[did you double-check that e.g. razers3 contains the openmp lib and e.g. doesn't?]

**Comment by marehr** [comment]
Nop, but the more dlls the merrier :D

**Comment by h-2** [comment]
@marehr  please do!

**Comment by marehr** [comment]
As you command sir, YES SIR YES!

```console
KLETTE+seqan_dev@KLETTE MINGW64 /c/Users/marehr/build_icc_package_alf/bin
$ unzip alf-1.1.5-Windows-AMD64.zip
Archive:  alf-1.1.5-Windows-AMD64.zip
   creating: alf-1.1.5-Windows-AMD64/bin/
  inflating: alf-1.1.5-Windows-AMD64/bin/alf.exe
  inflating: alf-1.1.5-Windows-AMD64/bin/api-ms-win-core-console-l1-1-0.dll
  inflating: alf-1.1.5-Windows-AMD64/bin/api-ms-win-core-datetime-l1-1-0.dll
  inflating: alf-1.1.5-Windows-AMD64/bin/api-ms-win-core-debug-l1-1-0.dll
  inflating: alf-1.1.5-Windows-AMD64/bin/api-ms-win-core-errorhandling-l1-1-0.dll
  inflating: alf-1.1.5-Windows-AMD64/bin/api-ms-win-core-file-l1-1-0.dll
  inflating: alf-1.1.5-Windows-AMD64/bin/api-ms-win-core-file-l1-2-0.dll
  inflating: alf-1.1.5-Windows-AMD64/bin/api-ms-win-core-file-l2-1-0.dll
  inflating: alf-1.1.5-Windows-AMD64/bin/api-ms-win-core-handle-l1-1-0.dll
  inflating: alf-1.1.5-Windows-AMD64/bin/api-ms-win-core-heap-l1-1-0.dll
  inflating: alf-1.1.5-Windows-AMD64/bin/api-ms-win-core-interlocked-l1-1-0.dll
  inflating: alf-1.1.5-Windows-AMD64/bin/api-ms-win-core-libraryloader-l1-1-0.dll
  inflating: alf-1.1.5-Windows-AMD64/bin/api-ms-win-core-localization-l1-2-0.dll
  inflating: alf-1.1.5-Windows-AMD64/bin/api-ms-win-core-memory-l1-1-0.dll
  inflating: alf-1.1.5-Windows-AMD64/bin/api-ms-win-core-namedpipe-l1-1-0.dll
  inflating: alf-1.1.5-Windows-AMD64/bin/api-ms-win-core-processenvironment-l1-1-0.dll
  inflating: alf-1.1.5-Windows-AMD64/bin/api-ms-win-core-processthreads-l1-1-0.dll
  inflating: alf-1.1.5-Windows-AMD64/bin/api-ms-win-core-processthreads-l1-1-1.dll
  inflating: alf-1.1.5-Windows-AMD64/bin/api-ms-win-core-profile-l1-1-0.dll
  inflating: alf-1.1.5-Windows-AMD64/bin/api-ms-win-core-rtlsupport-l1-1-0.dll
  inflating: alf-1.1.5-Windows-AMD64/bin/api-ms-win-core-string-l1-1-0.dll
  inflating: alf-1.1.5-Windows-AMD64/bin/api-ms-win-core-synch-l1-1-0.dll
  inflating: alf-1.1.5-Windows-AMD64/bin/api-ms-win-core-synch-l1-2-0.dll
  inflating: alf-1.1.5-Windows-AMD64/bin/api-ms-win-core-sysinfo-l1-1-0.dll
  inflating: alf-1.1.5-Windows-AMD64/bin/api-ms-win-core-timezone-l1-1-0.dll
  inflating: alf-1.1.5-Windows-AMD64/bin/api-ms-win-core-util-l1-1-0.dll
  inflating: alf-1.1.5-Windows-AMD64/bin/api-ms-win-crt-conio-l1-1-0.dll
  inflating: alf-1.1.5-Windows-AMD64/bin/api-ms-win-crt-convert-l1-1-0.dll
  inflating: alf-1.1.5-Windows-AMD64/bin/api-ms-win-crt-environment-l1-1-0.dll
  inflating: alf-1.1.5-Windows-AMD64/bin/api-ms-win-crt-filesystem-l1-1-0.dll
  inflating: alf-1.1.5-Windows-AMD64/bin/api-ms-win-crt-heap-l1-1-0.dll
  inflating: alf-1.1.5-Windows-AMD64/bin/api-ms-win-crt-locale-l1-1-0.dll
  inflating: alf-1.1.5-Windows-AMD64/bin/api-ms-win-crt-math-l1-1-0.dll
  inflating: alf-1.1.5-Windows-AMD64/bin/api-ms-win-crt-multibyte-l1-1-0.dll
  inflating: alf-1.1.5-Windows-AMD64/bin/api-ms-win-crt-private-l1-1-0.dll
  inflating: alf-1.1.5-Windows-AMD64/bin/api-ms-win-crt-process-l1-1-0.dll
  inflating: alf-1.1.5-Windows-AMD64/bin/api-ms-win-crt-runtime-l1-1-0.dll
  inflating: alf-1.1.5-Windows-AMD64/bin/api-ms-win-crt-stdio-l1-1-0.dll
  inflating: alf-1.1.5-Windows-AMD64/bin/api-ms-win-crt-string-l1-1-0.dll
  inflating: alf-1.1.5-Windows-AMD64/bin/api-ms-win-crt-time-l1-1-0.dll
  inflating: alf-1.1.5-Windows-AMD64/bin/api-ms-win-crt-utility-l1-1-0.dll
  inflating: alf-1.1.5-Windows-AMD64/bin/libmmd.dll
  inflating: alf-1.1.5-Windows-AMD64/bin/msvcp140.dll
  inflating: alf-1.1.5-Windows-AMD64/bin/ucrtbase.dll
  inflating: alf-1.1.5-Windows-AMD64/bin/vcruntime140.dll
   creating: alf-1.1.5-Windows-AMD64/example/
  inflating: alf-1.1.5-Windows-AMD64/example/small.fasta
  inflating: alf-1.1.5-Windows-AMD64/LICENSE
  inflating: alf-1.1.5-Windows-AMD64/README
```

```console
KLETTE+seqan_dev@KLETTE MINGW64 /c/Users/marehr/build_icc_package/bin
$ unzip razers3-3.5.3-Windows-AMD64.zip
Archive:  razers3-3.5.3-Windows-AMD64.zip
   creating: razers3-3.5.3-Windows-AMD64/bin/
  inflating: razers3-3.5.3-Windows-AMD64/bin/api-ms-win-core-console-l1-1-0.dll
  inflating: razers3-3.5.3-Windows-AMD64/bin/api-ms-win-core-datetime-l1-1-0.dll
  inflating: razers3-3.5.3-Windows-AMD64/bin/api-ms-win-core-debug-l1-1-0.dll
  inflating: razers3-3.5.3-Windows-AMD64/bin/api-ms-win-core-errorhandling-l1-1-0.dll
  inflating: razers3-3.5.3-Windows-AMD64/bin/api-ms-win-core-file-l1-1-0.dll
  inflating: razers3-3.5.3-Windows-AMD64/bin/api-ms-win-core-file-l1-2-0.dll
  inflating: razers3-3.5.3-Windows-AMD64/bin/api-ms-win-core-file-l2-1-0.dll
  inflating: razers3-3.5.3-Windows-AMD64/bin/api-ms-win-core-handle-l1-1-0.dll
  inflating: razers3-3.5.3-Windows-AMD64/bin/api-ms-win-core-heap-l1-1-0.dll
  inflating: razers3-3.5.3-Windows-AMD64/bin/api-ms-win-core-interlocked-l1-1-0.dll
  inflating: razers3-3.5.3-Windows-AMD64/bin/api-ms-win-core-libraryloader-l1-1-0.dll
  inflating: razers3-3.5.3-Windows-AMD64/bin/api-ms-win-core-localization-l1-2-0.dll
  inflating: razers3-3.5.3-Windows-AMD64/bin/api-ms-win-core-memory-l1-1-0.dll
  inflating: razers3-3.5.3-Windows-AMD64/bin/api-ms-win-core-namedpipe-l1-1-0.dll
  inflating: razers3-3.5.3-Windows-AMD64/bin/api-ms-win-core-processenvironment-l1-1-0.dll
  inflating: razers3-3.5.3-Windows-AMD64/bin/api-ms-win-core-processthreads-l1-1-0.dll
  inflating: razers3-3.5.3-Windows-AMD64/bin/api-ms-win-core-processthreads-l1-1-1.dll
  inflating: razers3-3.5.3-Windows-AMD64/bin/api-ms-win-core-profile-l1-1-0.dll
  inflating: razers3-3.5.3-Windows-AMD64/bin/api-ms-win-core-rtlsupport-l1-1-0.dll
  inflating: razers3-3.5.3-Windows-AMD64/bin/api-ms-win-core-string-l1-1-0.dll
  inflating: razers3-3.5.3-Windows-AMD64/bin/api-ms-win-core-synch-l1-1-0.dll
  inflating: razers3-3.5.3-Windows-AMD64/bin/api-ms-win-core-synch-l1-2-0.dll
  inflating: razers3-3.5.3-Windows-AMD64/bin/api-ms-win-core-sysinfo-l1-1-0.dll
  inflating: razers3-3.5.3-Windows-AMD64/bin/api-ms-win-core-timezone-l1-1-0.dll
  inflating: razers3-3.5.3-Windows-AMD64/bin/api-ms-win-core-util-l1-1-0.dll
  inflating: razers3-3.5.3-Windows-AMD64/bin/api-ms-win-crt-conio-l1-1-0.dll
  inflating: razers3-3.5.3-Windows-AMD64/bin/api-ms-win-crt-convert-l1-1-0.dll
  inflating: razers3-3.5.3-Windows-AMD64/bin/api-ms-win-crt-environment-l1-1-0.dll
  inflating: razers3-3.5.3-Windows-AMD64/bin/api-ms-win-crt-filesystem-l1-1-0.dll
  inflating: razers3-3.5.3-Windows-AMD64/bin/api-ms-win-crt-heap-l1-1-0.dll
  inflating: razers3-3.5.3-Windows-AMD64/bin/api-ms-win-crt-locale-l1-1-0.dll
  inflating: razers3-3.5.3-Windows-AMD64/bin/api-ms-win-crt-math-l1-1-0.dll
  inflating: razers3-3.5.3-Windows-AMD64/bin/api-ms-win-crt-multibyte-l1-1-0.dll
  inflating: razers3-3.5.3-Windows-AMD64/bin/api-ms-win-crt-private-l1-1-0.dll
  inflating: razers3-3.5.3-Windows-AMD64/bin/api-ms-win-crt-process-l1-1-0.dll
  inflating: razers3-3.5.3-Windows-AMD64/bin/api-ms-win-crt-runtime-l1-1-0.dll
  inflating: razers3-3.5.3-Windows-AMD64/bin/api-ms-win-crt-stdio-l1-1-0.dll
  inflating: razers3-3.5.3-Windows-AMD64/bin/api-ms-win-crt-string-l1-1-0.dll
  inflating: razers3-3.5.3-Windows-AMD64/bin/api-ms-win-crt-time-l1-1-0.dll
  inflating: razers3-3.5.3-Windows-AMD64/bin/api-ms-win-crt-utility-l1-1-0.dll
  inflating: razers3-3.5.3-Windows-AMD64/bin/libiomp5md.dll
  inflating: razers3-3.5.3-Windows-AMD64/bin/libmmd.dll
  inflating: razers3-3.5.3-Windows-AMD64/bin/msvcp140.dll
  inflating: razers3-3.5.3-Windows-AMD64/bin/razers3.exe
  inflating: razers3-3.5.3-Windows-AMD64/bin/ucrtbase.dll
  inflating: razers3-3.5.3-Windows-AMD64/bin/vcomp140.dll
  inflating: razers3-3.5.3-Windows-AMD64/bin/vcruntime140.dll
   creating: razers3-3.5.3-Windows-AMD64/example/
  inflating: razers3-3.5.3-Windows-AMD64/example/genome.fa
  inflating: razers3-3.5.3-Windows-AMD64/example/reads.fa
  inflating: razers3-3.5.3-Windows-AMD64/example/reads2.fa
  inflating: razers3-3.5.3-Windows-AMD64/LICENSE
  inflating: razers3-3.5.3-Windows-AMD64/README
```

**Comment by h-2** [comment]
:+1:

**Pull request merged**